server/sonar-server/src/main/java/org/sonar/server/computation/scm/ScmInfoImpl.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.scm;
+
+import com.google.common.base.Predicate;
+import java.util.Arrays;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.FluentIterable.from;
+import static com.google.common.collect.Iterables.isEmpty;
+import static java.util.Arrays.asList;
+
+@Immutable
+public class ScmInfoImpl implements ScmInfo {
+
+  @CheckForNull
+  private final Changeset latestChangeset;
+  private final Changeset[] lineChangesets;
+
+  public ScmInfoImpl(Iterable<Changeset> lineChangesets) {
+    this.lineChangesets = from(lineChangesets).toArray(Changeset.class);
+    this.latestChangeset = computeLatestChangeset(lineChangesets);
+  }
+
+  private static Changeset computeLatestChangeset(Iterable<Changeset> lineChangesets) {
+    checkState(!isEmpty(lineChangesets), "There's no changeset");
```

Comment:
I would move that check (and the use of CheckNotNull.INSTANCE) to the constructor before assigning the lineChangeset property (will save creating an array) and change the message to "A ScmInfo must have at least one Changeset and does not support any null one"